h/tasks/moderation.py

```diff
@@ -0,0 +1,96 @@
+from datetime import datetime
+
+from sqlalchemy import select
+
+from h.models import ModerationLog, ModerationStatus
+from h.models.notification import NotificationType
+from h.tasks.celery import celery, get_task_logger
+from h.services.notification import NotificationService
+
+log = get_task_logger(__name__)
+
+
+@celery.task
+def send_moderation_email(annotation_id, moderation_datetime_iso: str):
+    db = celery.request.db
+
+    print("send_moderation_email", annotation_id, moderation_datetime_iso)
+    moderation_datetime = datetime.fromisoformat(moderation_datetime_iso)
+
+    moderation_log = db.scalars(
+        select(ModerationLog)
+        .where(
+            ModerationLog.annotation_id == annotation_id,
+            ModerationLog.created >= moderation_datetime,
+            ModerationLog.notification_id.is_(None),
+        )
+        .order_by(ModerationLog.created.asc())
+        .with_for_update(skip_locked=True)
+    ).all()
+
+    if not moderation_log:
+        log.info(
+            "No moderation log found for annotation %s after %s",
+            annotation_id,
+            moderation_datetime,
+        )
+        return
+
+    if moderation_log[-1].created > moderation_datetime:
+        log.info(
+            "Moderation changes made after %s for annotation %s",
+            annotation_id,
+            moderation_datetime,
+        )
+        return
+
+    old_status = moderation_log[0].old_moderation_status
+    new_status = moderation_log[-1].new_moderation_status
+    print(moderation_log[0].id)
+    print(moderation_log[-1].id)
+    print(old_status, new_status)
+
+    if old_status == new_status:
```

Comment:
No change between the first and last moderation logs.

Reply:
~~While we don't want to send a notification for these we do want to exclude them from processing next time:~~

~~Let's say in quick succession:~~

~APPROVED -> DENIED -> APPROVED~

~Then 10 minutes later~

~DENIED~

~~Although  the result might be the same we only want take into account the last "APPROVED" in the logs, not start from the first one for the initial state.~~


I'm not 100% sure of this